### PR TITLE
Restore DefaultServlet.doPost to the previous major releases' behavior

### LINKED ISSUE #11299: EE8/9 `DefaultServlet.doPost()` doesn't behave like Jetty 10/11
**Jetty version(s)**
12

**Jetty Environment**
EE 8/9

**Description**
In Jetty 10/11 `DefaultServlet.doPost()` does what `doGet()` does:

```java
@Override
protected void doPost(HttpServletRequest request, HttpServletResponse response)
    throws ServletException, IOException
{
    doGet(request, response);
}
```

While  in Jetty 12 EE 8/9 it returns an error 405:

```java
@Override
protected void doPost(HttpServletRequest request, HttpServletResponse response)
    throws ServletException, IOException
{
    response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
}
```

The pre-existing behavior should be restored for backward compatibility.

### PR DESCRIPTION
**Pull request by lorban**
Only impacts EE 8/9.

Fixes #11299

### CODE REVIEW

File: jetty-ee9/jetty-ee9-servlet/src/main/java/org/eclipse/jetty/ee9/servlet/DefaultServlet.java [hunk 1]
```diff
@@ -514,7 +514,7 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
     protected void doPost(HttpServletRequest request, HttpServletResponse response)
         throws ServletException, IOException
     {
-        response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
+        doGet(request, response);
```
Comment (joakime) [inline jetty-ee9/jetty-ee9-servlet/src/main/java/org/eclipse/jetty/ee9/servlet/DefaultServlet.java:517]: the doOptions needs to be updated too (to include POST in its output)

**Review by joakime — CHANGES_REQUESTED** [review]

**Comment by joakime** [comment]
The doHead() method looks fishy here.
Do we have unit tests for it on ee9?

File: jetty-ee9/jetty-ee9-servlet/src/main/java/org/eclipse/jetty/ee9/servlet/DefaultServlet.java (hunk 1 above)
Comment (lorban) [reply jetty-ee9/jetty-ee9-servlet/src/main/java/org/eclipse/jetty/ee9/servlet/DefaultServlet.java:517]: Good catch!

**Comment by lorban** [comment]
@joakime we do have EE9 tests for `doHead` that make sure the content is empty, and I've also modified the `doPost` test to check that the content is returned.

**Comment by joakime** [comment]
@lorban you have OSGI test failures in both JVMs with this PR, how?? *boggle*

Try merging from `origin/jetty-12.0.x` HEAD and rebuild.

**Comment by lorban** [comment]
@joakime I think the OSGi tests are flaky as I've seen them failing in other PRs, and re-running the build on this one made them turn green.

**Review by joakime — APPROVED** [review]
Merge it!

**Pull request merged**